OpenRA.Mods.Common/Traits/Render/RenderRangeCircle.cs

```diff
@@ -32,7 +32,7 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr, World w, ActorInfo ai,
 			var range = FallbackRange;
 
 			if (armaments.Any())
-				range = armaments.Select(a => w.Map.Rules.Weapons[a.Weapon.ToLowerInvariant()].Range).Max();
+				range = armaments.Select(a => w.Map.Rules.Weapons[a.Weapon.ToLowerInvariant()].DefaultModifiedRange(ai.Traits.WithInterface<IRangeModifierInfo>())).Max();
```

Comment:
You'll probably want to cache ```ai.Traits.WithInterface<IRangeModifierInfo>())``` outside the ```Select``` so it's not repeated per armament. It would also help make this line less long!